### ISSUE TITLE
Insert to bucketed and sorted Hive table produces not sorted files on file system

### ISSUE THREAD

**Issue by OleksiiDuzhyi** [trinodb/trino#2450]
Hi,

I've created a bucketed and sorted table in Hive plugin: 
```sql
CREATE TABLE hive.default.sorted_parquet (uuid varchar, id integer ) 
WITH ( 
format = 'PARQUET', 
bucketed_by=ARRAY['uuid'],
bucket_count=2, 
sorted_by = ARRAY['uuid']
);
```

After that I made a couple of inserts of fairly large sets of data with session `parquet_writer_block_size` fairly small to produce parquet files with at least 2 blocks. Analyzing one of the outcome files with `parquet-tools` shows following information for `uuid` column:

```
Block 0
Min: 0d6fb314-aec3-4447-ad9d-a4558f5bf351 
Max: fda7f909-661d-4ceb-a08b-d0b44490fe59

Block 1
Min: 01107f6c-f212-4c97-8d97-f3992151de99
Max: f51cded7-aee3-4651-9cdc-68b53a4c0684
```
During write - temporary files are not created as well. Only `CREATE TABLE ... AS SELECT ...` works as expected - first creating tmp files for each split, and then merging them back to sorted file.

This indicates that data in the parquet files is not sorted inside and with `ORDER BY` optimization turned on - provides no mechanism to guarantee physical order of data during non-initial insert of data.

**Comment by sopel39** [comment]
This seems like a bug

**Comment by l1x** [comment]
I don't think there is sorting on writes:

https://github.com/prestodb/presto/issues/6666#issuecomment-283849259

**Comment by OleksiiDuzhyi** [comment]
Bug was opened based on this slack discussion - https://prestosql.slack.com/archives/CGB0QHWSW/p1573745523046400 . 

There is no way to guarantee on disk data sorting during insert with latest `optimizer.skip-redundant-sort` optimization if sort on writes is not supported.

**Comment by ddrinka** [comment]
Is this closed by https://github.com/prestosql/presto/pull/1870 for you?  I'm able to verify sorted data on disk with the current codebase, while using Glue...

**Comment by raunaqmorarka** [comment]
Going to assume this is fixed by #1870 , feel free to reopen if that's not the case
